### ISSUE TITLE
Bug: `no-unsafe-values` does not report JSON5 `Infinity` values

### ISSUE THREAD

**Issue by electrohyun** [eslint/json#277]
### Environment

ESLint version: v10.9.1
@eslint/json version: v2.1.0
Node version: v22.22.3
npm version: v10.8.2
Operating System: Ubuntu 20.04.0 LTS (StackBlitz)

### Which language are you using?

json5

### What did you do?

<details>
<summary>Configuration</summary>

```js
import json from "@eslint/json";

export default [
  {
    files: ["**/*.json5"],
    plugins: { json },
    language: "json/json5",
    rules: { "json/no-unsafe-values": "error" },
  },
];
```

</details>

```json5
[2e308, +2e308, -2e308, Infinity, +Infinity, -Infinity]
```

### What did you expect to happen?

All six values should be reported.

### What actually happened?

<img width="2444" height="1912" alt="Image" src="https://github.com/user-attachments/assets/b5ba1770-7437-484a-9af7-537faebc6453" />

Only 2e308, +2e308, and -2e308 are reported.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-jgyzylpm?file=index.js

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

I used an AI to correct the grammar in the section below to make it clearer.

### Additional comments

This is something I thought about while working on #266, and after looking into it, it appears to be a bug. If you agree, I’d be happy to submit a PR right away.

<hr />

Disclosure: I'm a participant of [open source contribution program OSSCA](https://github.com/eslint-ossca)

**Comment by DMartens** [comment]
Thank you for the issue.
I think this is an oversight as JSON5 allows `Infinity` while the default specification does not.
A PR for this is welcome.
